src/cls/rgw/cls_rgw_ops.h

```diff
@@ -32,26 +32,20 @@ struct rgw_cls_obj_prepare_op
   cls_rgw_obj_key key;
   std::string tag;
   std::string locator;
-  bool log_op;
-  uint16_t bilog_flags;
-  rgw_zone_set zones_trace;
 
-  rgw_cls_obj_prepare_op() : op(CLS_RGW_OP_UNKNOWN), log_op(false), bilog_flags(0) {}
+  rgw_cls_obj_prepare_op() : op(CLS_RGW_OP_UNKNOWN) {}
 
   void encode(ceph::buffer::list &bl) const {
-    ENCODE_START(7, 5, bl);
+    ENCODE_START(8, 5, bl);
     uint8_t c = (uint8_t)op;
     encode(c, bl);
     encode(tag, bl);
     encode(locator, bl);
-    encode(log_op, bl);
     encode(key, bl);
-    encode(bilog_flags, bl);
-    encode(zones_trace, bl);
```

Comment:
removing fields from the v8 encoding will break v7 decoders that still expect those fields to be present

we'll need to continue encoding empty values for these to preserve compatibility

Reply:
I see; then how can we drop them? I mean, if we don't drop them in encoding, perhaps we need to keep them in encoding as well. So what should we do in these cases?

Reply:
probably easiest to leave `struct rgw_cls_obj_prepare_op` and its encodings unchanged. your update to `cls_rgw_bucket_prepare_op()` will just leave them empty

Reply:
Can't we perhaps play with vars like `require_osd_release` in osdmap to make sure all OSDs are on encoding 8 at least so we stop encoding?

Reply:
@cbodley as discussed, the encoding and decoding are kept as it was to maintain compatibility now.